src/main/scala/rps/GameRepository.scala

```diff
@@ -1,40 +1,43 @@
 package rps
 
-import scala.concurrent.{ExecutionContext, Future}
+import scala.concurrent.ExecutionContext
 import java.util.UUID
 import java.sql.Timestamp
 import java.time.Instant
 
 import slick.driver.H2Driver.backend.DatabaseDef
 import slick.driver.H2Driver.api._
-
-import db.Tables.{Plays, PlayRow}
+import db.Tables.{PlayRow, Plays}
 import model._
+import zio.{IO, Task}
 
 trait GameRepository {
-  def save(play: Play): Future[Either[Throwable, UUID]]
-  def read(): Future[Either[Throwable, Option[Play]]]
+  def save(play: Play): IO[Throwable, UUID]
+  def read(): IO[Throwable, Option[Play]]
```

Comment:
```suggestion
  def save(play: Play): Task[UUID]
  def read(): Task[Option[Play]]
```

Reply:
Since this will be a tutorial, I didn't want to introduce too many names and avoided Task... also, we could put there some domain-specific errors instead of Throwable

Reply:
👍 makes sense, let's then introduce a domain-specific error